### PR TITLE
hosts paginated list

### PR DESCRIPTION
**Pull request by kgaikwad**
Required backend side [PR-18](https://github.com/RedHatInsights/ros-backend/pull/18)

### CODE REVIEW

**Comment by kgaikwad** [comment]
Attaching a screenshot for reference:

Screenshot from 2021-02-02 21-35-09

**Comment by kgaikwad** [comment]
With more records -
Screenshot from 2021-02-03 00-38-28

**Comment by kgaikwad** [comment]
+ adding @karelhala into loop for review.

File: src/Routes/RosPage/RosPage.js [hunk 1]
```diff
@@ -21,14 +24,53 @@ class RosPage extends React.Component {
...
+        const { perPage } = this.state;
+        const pagination = { page, perPage };
+        this.updatePagination(pagination);
+    }
+
+    onPerPageSelect(event, perPage) {
+        const page = 1;
+        const pagination = { page, perPage };
+        this.updatePagination(pagination);
+    }
```
Comment (karelhala) [inline src/Routes/RosPage/RosPage.js:52]: You don't really need these 2 functions, you can call directly `updatePagination`

When setting page: 
```JSX
onSetPage={ (_e, page) => this.updatePagination({ page, perPage: this.state.perPage })}
```
When setting perPage:
```JSX
onPerPageSelect={(_e, perPage) => this.updatePagination({ page: 1, perPage })}
```

File: src/Routes/RosPage/RosPage.js [hunk 2]
```diff
@@ -21,14 +24,53 @@ class RosPage extends React.Component {
...
+
+    updatePagination(pagination) {
+        this.setState({ page: pagination.page, perPage: pagination.perPage });
+        this.fetchWithParams({ page: pagination.page, perPage: pagination.perPage });
```
Comment (karelhala) [inline src/Routes/RosPage/RosPage.js:56]: If you are going to follow the same shape, no need to re-assign. And would call directly the `fetchSystems` function.
```suggestion
        this.setState(pagination);
        fetchSystems(pagination);
```

File: src/Routes/RosPage/RosPage.js [hunk 3]
```diff
@@ -21,14 +24,53 @@ class RosPage extends React.Component {
...
+        };
+
+        this.onSetPage = this.onSetPage.bind(this);
+        this.onPerPageSelect = this.onPerPageSelect.bind(this);
     }
 
     async componentDidMount() {
         await window.insights.chrome.auth.getUser();
-        this.props.fetchSystems();
+        this.fetchWithParams();
```
Comment (karelhala) [inline src/Routes/RosPage/RosPage.js:39]: I would call directly the dispatched fetch function with state. `fetchSystems({ page: this.state.page, perPage: this.state.perPage })`

**Review by karelhala — COMMENTED** [review]
Looking good, just a few improvements to make this more easy to read, unless you want to add some other things when fetching systems no need to have custom function that just merges state and passed in props.

**Review by terezanovotna — APPROVED** [review]
Pagination looking good! Thanks Kavita.

**Comment by kgaikwad** [comment]
@karelhala, 
Updated changes as per your suggestions. Also used shared components i.e PrimaryToolbar, TableToolbar.
Please request to review updated changes.

**Comment by kgaikwad** [comment]
Backend side PR https://github.com/RedHatInsights/ros-backend/pull/18 is merged.

File: src/Routes/RosPage/RosPage.js [hunk 4]
```diff
@@ -37,9 +54,30 @@ class RosPage extends React.Component {
                 <Main>
                     <Card className='pf-t-light  pf-m-opaque-100'>
                         <CardBody>
-                            <div>
-                                { (!this.props.loading) ? (<RosTable systems = { this.props.systemsData }/>) : null }
-                            </div>
+                            <React.Fragment>
```
Comment (karelhala) [inline src/Routes/RosPage/RosPage.js:57]: I don't think you need to wrap this in `React.Fragment`

**Review by karelhala — APPROVED** [review]

**Comment by kgaikwad** [comment]
Thank you so much @karelhala & @terezanovotna for review and suggestions.

**Pull request merged**